### PR TITLE
A Semi-non-linear Bastion Factory

### PR DESCRIPTION
**Pull request by ribagi**
New items from a few unused factories. Walls from Fancy Ore semlter, Base from Smelter, Gearbox from Iron Forge, Object d'art from Stain Glass, Flooring from Stone Brick, Rations from Bakery, Smaragdus from Iron Cauldron, Silicon Tetranitratobihydrotrioxycarbon from Explosive, and Pure Ice from Crystallisation Factory

### CODE REVIEW

**Comment by ttk2** [comment]
and you put all of those together for the bastion factory?

**Comment by ribagi** [comment]
Yea

**Comment by pruby** [comment]
I really like the direction here - collecting inputs from the other factories. It provides the difficulty beyond cost that my recipe lacks. However, it seems way too cheap if you already have the other factories, which a lot of towns do. Total cost 48d in diamonds, 4d of exp, then a few other bits and pieces which at an extreme stretch could bring it up to 64d, then 1d per bastion.  In my opinion it needs to be much higher than that, by about 8-10 times.

Also need to touch up indentation in a couple of places. YAML is indentation sensitive - keys at the same level _must_ have the same indentation. In particular "outputs" needs to be on the same indentation as "inputs", and line 6835 looks indented too far (next line should be indented, is on same level).

**Comment by ribagi** [comment]
The costs I pulled out of the air. Even though when I did put them out, I did feel that they were on the low side. It is also on a side note that the cheaper items are only there to server as a check if you have a factory or not. The methodology on when that happens is quite random.

On the formatting thing, one thing I hate about notepad --> github is that while it looks perfect in notepad, when you send a pull request everything goes to shit.

**Comment by ttk2** [comment]
so up the cost, get it formatted and then merge this one it is?

**Comment by pruby** [comment]
Still tabs mixed through this - will patch and add extra costs.

**Comment by ribagi** [comment]
x8 item 'Base' cost, x4 item 'Walls' cost, and x2 item 'Smaragdus' cost to the Final Bastion Factory.

The Emerald Block cost to Smaragdus increased by x3.

Some minor fixes too.

**Comment by pruby** [comment]
Looks like we were amending at the same time - added a version on https://github.com/ttk2/FactoryMod/pull/21 with docs & cost breakdown. I've increased the costs _way_ further than that.

File: config.yml [hunk 1]
```diff
@@ -2288,6 +2297,68 @@ production_factories:
...
       Stick:
         material: STICK
+  Bastion_Factory:
+    name: Bastion Factory
+    fuel:
+      Charcoal:
+        material: COAL
+        durability: 1
+    inputs:
+	  Silicon Tetranitratobihydrotrioxycarbon:
```
Comment (pruby) [inline config.yml:2307]: These lines are still tab-indented.

File: config.yml [hunk 2]
```diff
@@ -6542,4 +6613,277 @@ production_recipes:
...
+        material: INK_SACK
+        amount: 4
+        durablity: 4
+        display_name: Walls
+        lore: An item used to create a Bastion Block
+    outputs:
+      Sponge:
+        material: SPONGE
+        amount: 64     
+
```
Comment (pruby) [inline config.yml:6889]: And a final tab here. They're dangerous in YAML because you can't see the difference, but the parser does.

**Pull request closed without merging**